### DISCUSSION TITLE
theme type

### DISCUSSION THREAD

**Discussion by zoy-l** [emotion-js/emotion#2881] [Ideas]
an idea ~

Is it possible to extend theme type by `import(to-key).Theme` and then enter `to-key` in tsconfig paths, final
```
export type Theme = {...}
// or 
// import xxx from 'source'
// exprot Theme = typeof xxx
```

this avoids the need to manually enter the type and can be deduced

**Comment by srmagura** [comment]
This section in our docs shows how to define a theme type: https://emotion.sh/docs/typescript#define-a-theme

You should be able to do something like this:

```ts
import MyTheme from 'source';

declare module '@emotion/react' {
  export type Theme = typeof MyTheme
}
```

Whether you need `typeof` depends on whether `MyTheme` is an object or a TypeScript type.

**Comment by zoy-l** [comment]
thanks for the reply, yes, I made a mistake, I always thought that global types cannot be `import 'source'` 😅
